grug/app/src/app.rs

```diff
@@ -48,21 +51,23 @@ pub struct App<DB, VM, PP = NaiveProposalPreparer> {
     query_gas_limit: u64,
 }
 
-impl<DB, VM, PP> App<DB, VM, PP> {
-    pub fn new(db: DB, vm: VM, pp: PP, query_gas_limit: u64) -> Self {
+impl<DB, VM, PP, ID> App<DB, VM, PP, ID> {
+    pub fn new(db: DB, vm: VM, pp: PP, indexer: ID, query_gas_limit: u64) -> Self {
         Self {
             db,
             vm,
             pp,
+            indexer,
             query_gas_limit,
         }
     }
 }
 
-impl<DB, VM, PP> App<DB, VM, PP>
+impl<DB, VM, PP, ID> App<DB, VM, PP, ID>
 where
     DB: Db,
     VM: Vm + Clone,
+    ID: Indexer,
     PP: ProposalPreparer,
     AppError: From<DB::Error> + From<VM::Error> + From<PP::Error>,
```

Comment:
Similar to DB, VM, PP, add the following bound: `AppError: From<ID::Error>`. This allow to use `?` instead of `unwrap` for `Indexer` functions.